cmake/multi_image.cmake

```diff
@@ -38,23 +38,16 @@ function(image_board_selection board_in board_out)
   # images defined.
   # TODO: Allow multiple non-secure images by using Kconfig to set the
   # secure/non-secure property rather than using a separate board definition.
-  set(nonsecure_boards_with_ns_suffix
-    nrf9160_pca10090ns
-    nrf9160_pca20035ns
-    nrf5340_dk_nrf5340_cpuappns
-    )
-  if(${board_in} IN_LIST nonsecure_boards_with_ns_suffix)
-    string(LENGTH ${board_in} len)
-    math(EXPR len_without_suffix "${len} - 2")
-    string(SUBSTRING ${board_in} 0 ${len_without_suffix} board_in_without_suffix)
-
+  string(REGEX REPLACE "(_?ns)$" "" board_in_without_suffix ${board_in})
+  if(NOT ${board_in} STREQUAL ${board_in_without_suffix})
+    if (NOT CONFIG_ARM_NONSECURE_FIRMWARE)
+      message(FATAL_ERROR "${board_in} is not a valid name for a board without "
+      "'CONFIG_ARM_NONSECURE_FIRMWARE' set. This because the 'ns'/'_ns' ending "
```

Comment:
"Non-secure firmware must use a board whose name ends with 'ns' or '_ns'."

Reply:
We are checking the opposite, whether boards which does not have non-secure capabilities have called themself something ending in _ns or ns

Reply:
Aa, sorry. 

"The board ${board_in} cannot be used when"
"'CONFIG_ARM_NONSECURE_FIRMWARE' is not set,"
"Board names ending with 'ns' or '_ns' are reserved for non-secure firmware."

Reply:
We must specify that it is the NAME of the board that is invalid, not the selected board itself, this is not clear from the first sentence in your proposal IMO.